pages/api/users/search.js

```diff
@@ -0,0 +1,54 @@
+import connectMongo from "@config/mongo";
+import logger from "@config/logger";
+import findAllBasic from "@services/profiles/findAllBasic";
+import { Profile } from "@models/index";
+
+
+export default async function handler(req, res) {
+  if (req.method != "GET") {
+    return res
+      .status(400)
+      .json({ error: "Invalid request: GET request required" });
+  }
+
+  await connectMongo();
+
+  let fileProfiles = findAllBasic();
+
+  const {} = req.query;
+  
+  let dbProfiles = [];
+  try {
+    dbProfiles = await Profile.find({ location: { $ne: null } });
+  } catch (e) {
+    logger.error(e, "failed loading profile from db");
+    return fileProfiles;
+  }
+
+  if(initialRandomUsers) {
+    fileProfiles = fileProfiles.slice(0, 0+5);
+  }
+
+  const profiles = fileProfiles.map((fileProfile) => {
+    const profile = dbProfiles.find(
+      (dbProfile) => dbProfile.username === fileProfile.username
+    );
+
+    const basicProfile = {
+      username: fileProfile.username,
+      name: fileProfile.name,
+      bio: fileProfile.bio,
+      tags: fileProfile.tags,
+    };
+
+    if (profile && profile._doc.location.name !== "unknown") {
+      return {
+        ...basicProfile,
+        location: profile._doc.location,
+      };
+    }
+    return basicProfile;
+  });
+
+  res.status(200).json(profiles);
+}
```

Comment:
Why is an extra api needed?

The current api can return
- list of users if no parameters
- random if query parameter set
- search if query parameter set

Reply:
Sorry this is a temporary file I forgot to delete. Fixed.